client/src/containers/AdvancedSearch/AdvancedSearch.js

```diff
@@ -864,6 +865,25 @@ export class AdvancedSearch extends Component {
       timer: 1500,
     });
 
+  addParamasInRechercher = (key, subitem) => {
+    let recherche = [...this.state.recherche];
+    recherche[key] = {
+      ...recherche[key],
+      value: subitem.name || subitem.formatted_address,
+      icon: subitem.icon,
+      query:
+        subitem.query ||
+        subitem.address_components ||
+        (key !== 3 ? subitem.name : undefined),
+      active: true,
+      ...(subitem.short && { short: subitem.short }),
+      ...(subitem.bottomValue && { bottomValue: subitem.bottomValue }),
+      ...(subitem.topValue && { topValue: subitem.topValue }),
+    };
+    this.setState({ recherche: recherche });
```

Comment:
tu peux juste mettre "this.setState({ recherche })" ca fait la meme chose que toi mais en plus court